### DISCUSSION TITLE
A couple of ideas for new features aimed at reducing bandwidth usage

### DISCUSSION THREAD

**Discussion by Cold-Lemonade** [blakeblackshear/frigate#15054] [Ideas]
@blakeblackshear @NickM-27 @hawkeye217 -- Thank you for developing such amazing NVR software. I really appreciate the work that you and others have been putting into Frigate. I have been running Frigate 0.14.1 flawlessly since it was released. However, I have a couple of observations and suggestions for how it can be improved further.

**1. Secondary GUI showing latest still frame from of each camera** (like the old one from ver. 0.13.2). The new GUI with it its live view is great, but it would be nice to have the option to also use an interface that demands less bandwidth. The new live view GUI is great if I'm on wifi where data usage limits are of no concern. However, when I'm looking at Frigate from my iPhone on a cellular network, it would be nice to have an interface that transmits less data. Would it be possible to add an option to display the old GUI from ver 0.13.2 at another port? So that users can use this GUI if they face data constraints?

**2. Record low res stream for smaller images/video in GUI**. The new GUI from ver 0.14.1 displays recordings in smaller size (e.g., when reviewing the history from one camera, the sync'ed recording from the other cameras appears below in smaller size). Would it be possible to have the option to record the sub stream and use the sub stream recording for these smaller images/video? I think this would reduce the data transmitted between Frigate and the client device.

**3. Display sub stream in GUI main page and high res stream for view when user expands on a single camera** (clicks on any given camera). If I am not using my iPhone to view the Frigate GUI, I am using a laptop. The laptop screen isn't large enough to benefit from viewing the high res streams in the GUI's main page. Instead, I think it would be better to have the option to display the sub stream and allow the user to set how frequently it updates (e.g., 1 per second, 2 per second, etc.).

I think others would benefit from these features. Thank you for considering them.

**Comment by NickM-27** [comment]
1. You can always disable the automatic live view behavior in the UI settings tab. By default in the dashboard the still frames are shown and they are actually more bandwidth efficient than the old UI. Also, if bandwidth is low then jsmpeg will be used instead of MSE after MSE can't load. In general, no, I don't think any effort will be put in to maintaining or doing anything with the old UI. 
2. There is a feature request for this, in general though I think the previews features covers this just fine. You can swipe to the right on preview items and it uses very little bandwidth to show what happened. 
3. This has already been implemented for 0.16

**Reply by hawkeye217** [reply]
To add on to what @NickM-27 said, in 0.16 you'll also be able to set a go2rtc stream and streaming method for each camera in your camera group's dashboard, per-device. So on desktops, you can have your default be the full resolution stream and on mobile, you can select a substream for any or all of your cameras.

**Reply by Cold-Lemonade** [reply]
In terms of selecting whether to display the high res or low res stream in the GUI, etc., will there be a way to avoid restreaming the high res stream using go2rtc? I have several 8MP cameras and restreaming them drives the cpu usage for go2rtc close to 30%. Because of the improvements in the Frigate GUI, I don't intend use the high res streams outside of Frigate.

**Reply by hawkeye217** [reply]
No, go2rtc will be required to use the new feature. But go2rtc won't use cpu for a stream if something isn't consuming it, so it you could still define the stream in go2rtc and just not consume it elsewhere in your frigate config.
